src/db/transaction.rs

```diff
@@ -122,7 +131,36 @@ impl DbTransactionPool {
         A: FnOnce(Box<dyn Db<'a>>) -> F,
         F: Future<Output = Result<HttpResponse, Error>> + 'a,
     {
-        let (resp, db) = self.transaction_internal(action).await?;
+        // Get connection from pool
+        let db = self.pool.get().await?;
+        let resource_ts = db
+            .extract_resource(
+                self.user_id.clone(),
+                self.collection.clone(),
+                self.bso_opt.clone(),
+            )
+            .await?;
```

Comment:
This should probably happen inside of the transaction.

Reply:
Ah yes. Thank you!